### ISSUE TITLE
Items are not added to the same cart they are ordered from

### ISSUE THREAD

**Issue by corey-aloia** [SAP/spartacus#9694]
Opening a ticket here to start the discussion. 
The following functionality in CDS is broken:
- Anonymous user adds an item to cart
- The user Registers + Logs in
- The user places an order

The CDS backend creates a json document whereby the properties for a cart and an order are connected.
This connection requires 2 events:
AddedToCart
OrderCreated

In this scenario, the AddedToCart event comes from Spartacus and contains the cartCode of the anonymous user. The OrderCreated event comes from the EC backend, and contains the cartCode of the logged-in user.

These two cartCodes are different, meaning CDS cannot connect an order to a cart.

The reason is because an anonymous user has a cartCode and a cart guid. When this user Registers + logs-in, they get a different cartCode and a different guid.

TLDR; items were added to cart '123' and ordered from cart '234'.

The occ documentation states that the guid should be used for an anonymous user, and the cartCode for a known user: https://help.sap.com/viewer/9d346683b0084da2938be8a285c0c27a/2005/en-US/8c425bb4866910149bf3c19b8f239309.html?q=%2Fanonymous%2Fcarts

But, anonymous and known users have both of these properties.

It isn't clear if this is an issue with OCC, or with Spartacus creating a new cart when it could potentially attach a cart to a known user

**Comment by ReneRath** [comment]
AFAICS in the accelerator storefront the AddedToCart event already contains the cartCode (integer as opposed of guid). So from the looks of it, Spartacus behaves different than Accelerator here. Cannot say which is more correct.

**Comment by Xymmer** [comment]
to be investigated @ReneRath can you contact me about priority?
